integration_tests/2pc_test.go

```diff
@@ -1742,3 +1743,90 @@ func (s *testCommitterSuite) TestFlagsInMemBufferMutations() {
 		s.Equal(assertNotExist, mutations.IsAssertNotExist(i))
 	})
 }
+
+func (s *testCommitterSuite) TestSetLockedKeyValue() {
+	ctx := context.Background()
+	k1 := []byte("k1")
+	v1 := []byte("v1")
+	v2 := []byte("v2")
+
+	mustLockKey := func(txn transaction.TxnProbe, key []byte) {
+		s.Require().NoError(txn.LockKeys(ctx, &kv.LockCtx{ForUpdateTS: txn.StartTS(), WaitStartTime: time.Now()}, key))
+	}
+	mutOpVals := func(opVals ...interface{}) func(m transaction.CommitterMutations) {
+		s.Require().Equal(0, len(opVals)%2)
+		return func(m transaction.CommitterMutations) {
+			s.Require().Equal(m.Len(), len(opVals)/2)
+			for i := 0; i < len(opVals); i += 2 {
+				s.Require().Equal(opVals[i], m.GetOp(0))
+				s.Require().Equal(opVals[i+1], m.GetValue(0))
+			}
+		}
+	}
+
+	for _, tt := range []struct {
+		name    string
+		actions []func(txn transaction.TxnProbe)
+		check   func(m transaction.CommitterMutations)
+	}{
+		{
+			"NoLock",
+			[]func(txn transaction.TxnProbe){
+				func(txn transaction.TxnProbe) { txn.SetLockedKeyValue(k1, v1) },
+			},
+			mutOpVals(),
+		},
+		{
+			"LockOnly",
+			[]func(txn transaction.TxnProbe){
+				func(txn transaction.TxnProbe) { txn.SetLockedKeyValue(k1, v1) },
+				func(txn transaction.TxnProbe) { mustLockKey(txn, k1) },
+			},
+			mutOpVals(kvrpcpb.Op_Put, v1),
+		},
+		{
+			"LockAndSet",
+			[]func(txn transaction.TxnProbe){
+				func(txn transaction.TxnProbe) { txn.SetLockedKeyValue(k1, v1) },
+				func(txn transaction.TxnProbe) { mustLockKey(txn, k1) },
+				func(txn transaction.TxnProbe) { s.Require().NoError(txn.Set(k1, v2)) },
+			},
+			mutOpVals(kvrpcpb.Op_Put, v2),
+		},
+		{
+			"LockAndDelete",
+			[]func(txn transaction.TxnProbe){
+				func(txn transaction.TxnProbe) { txn.SetLockedKeyValue(k1, v1) },
+				func(txn transaction.TxnProbe) { mustLockKey(txn, k1) },
+				func(txn transaction.TxnProbe) { s.Require().NoError(txn.Delete(k1)) },
+			},
+			mutOpVals(kvrpcpb.Op_Del, []byte{}),
+		},
```

Comment:
Is it necessary to test in different order, like calling `txn.SetLockedKeyValue` after calling `LockKeys` or any mutations?

Reply:
Ah, I submitted the  review to wrong place